code/modules/vehicles/_hitbox.dm

```diff
@@ -0,0 +1,223 @@
+/**
+ * HITBOX
+ * The core of multitile. Acts as a relay for damage and stops people from walking onto the multitle sprite
+ * has changed bounds and as thus must always be forcemoved so it doesnt break everything
+ * I would use pixel movement but the maptick caused by it is way too high and a fake tile based movement might work? but I want this to at least pretend to be generic
+ * Thus we just use this relay. it's an obj so we can make sure all the damage procs that work on root also work on the hitbox
+ */
+/obj/hitbox
+	density = TRUE
+	anchored = TRUE
+	invisibility = INVISIBILITY_MAXIMUM
+	bound_width = 96
+	bound_height = 96
+	bound_x = -32
+	bound_y = -32
+	max_integrity = INFINITY
+	move_resist = INFINITY // non forcemoving this could break gliding so lets just say no
+	///people riding on this hitbox that we want to move with us
+	var/list/atom/movable/tank_desants
+	///The "parent" that this hitbox is attached to and to whom it will relay damage
+	var/obj/vehicle/root = null
+
+/obj/hitbox/Initialize(mapload, obj/vehicle/new_root)
+	. = ..()
+	root = new_root
+	allow_pass_flags = root.allow_pass_flags
+	flags_atom = root.flags_atom
+	resistance_flags = root.resistance_flags
+	RegisterSignal(new_root, COMSIG_MOVABLE_MOVED, PROC_REF(root_move))
+	RegisterSignal(new_root, COMSIG_QDELETING, PROC_REF(root_delete))
+	RegisterSignals(new_root, list(COMSIG_RIDDEN_DRIVER_MOVE, COMSIG_VEHICLE_MOVE), PROC_REF(on_attempt_drive))
+	RegisterSignal(new_root, COMSIG_ATOM_DIR_CHANGE, PROC_REF(owner_turned))
+
+	var/static/list/connections = list(
+		COMSIG_OBJ_TRY_ALLOW_THROUGH = PROC_REF(can_cross_hitbox),
+		COMSIG_TURF_JUMP_ENDED_HERE = PROC_REF(on_jump_landed),
+		COMSIG_ATOM_EXITED = PROC_REF(on_exited),
+	)
+	AddElement(/datum/element/connect_loc, connections)
+
+/obj/hitbox/Destroy(force)
+	if(!force) // only when the parent is deleted
+		return QDEL_HINT_LETMELIVE
+	root?.hitbox = null
+	root = null
+	return ..()
+
+///signal handler for handling PASS_WALKOVER
+/obj/hitbox/proc/can_cross_hitbox(datum/source, atom/mover)
+	SIGNAL_HANDLER
+	if(locate(src) in mover.loc)
```

Comment:
not sure if it makes a diff perf wise, but you can check if mover.loc is in locs

Reply:
Yea heres the funny thing when I did it that way it only returned true when src.loc == mover.loc